llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -13273,22 +13281,34 @@ struct NodeExtensionHelper {
     return OrigOperand.getOpcode() == RISCVISD::VMV_V_X_VL;
   }
 
+  /// Get the extended opcode.
+  unsigned getExtOpc(ExtKind SupportsExt) const {
+    switch (SupportsExt) {
```

Comment:
Can we just remove `ExtKind` (that means we use `std::optional<int>`) and use `RISCVISD::VSEXT_VL`, `RISCVISD::VZEXT_VL` and `RISCVISD::FP_EXTEND_VL` directly?

Reply:
It might be more readable with a enum? We could know exactly what extensions are currently supported.